mongo/options/clientoptions.go

```diff
@@ -300,187 +322,305 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
+// operation, this is the acceptable non-negative delta between shortest and longest average round-trip times. Servers
+// within the latency. This can also be set through the "localThresholdMS" URI option (e.g. "localThresholdMS=15000").
+// The default is 15 milliseconds.
 func (c *ClientOptions) SetLocalThreshold(d time.Duration) *ClientOptions {
 	c.LocalThreshold = &d
 	return c
 }
 
-// SetMaxConnIdleTime specifies the maximum number of milliseconds that a connection can remain idle
-// in a connection pool before being removed and closed.
+// SetMaxConnIdleTime specifies the maximum amount of time that a connection can remain in a connection pool without
+// being used before it is removed and closed. This can also be set through the "maxIdleTimeMS" URI option (e.g.
+// "maxIdleTimeMS=10000"). The default is 0, meaning a connection can remain unused indefinitely.
 func (c *ClientOptions) SetMaxConnIdleTime(d time.Duration) *ClientOptions {
 	c.MaxConnIdleTime = &d
 	return c
 }
 
-// SetMaxPoolSize specifies the max size of a server's connection pool.
+// SetMaxPoolSize specifies the maximum number of connections allowed in each server's connection pool. Requests to a
+// server will block if this maximum is reached. This can also be set through the "maxPoolSize" URI option (e.g.
+// "maxPoolSize=100"). The default is 100. If this is 0, it will be set to math.MaxInt64.
 func (c *ClientOptions) SetMaxPoolSize(u uint64) *ClientOptions {
 	c.MaxPoolSize = &u
 	return c
 }
 
-// SetMinPoolSize specifies the min size of a server's connection pool.
+// SetMinPoolSize specifies the minimum number of connections allowed in each server's connection pool. If this is
+// non-zero, each server's pool will be maintained in the background to ensure that the size does not fall below the
+// minimum. This can also be set through the "minPoolSize" URI option (e.g. "minPoolSize=100"). The default is 0.
 func (c *ClientOptions) SetMinPoolSize(u uint64) *ClientOptions {
 	c.MinPoolSize = &u
 	return c
 }
 
-// SetPoolMonitor specifies the PoolMonitor for a server's connection pool.
+// SetPoolMonitor specifies a PoolMonitor to receive connection pool events. See the event.PoolMonitor documentation
+// for more information about the structure of the monitor and events that can be received.
 func (c *ClientOptions) SetPoolMonitor(m *event.PoolMonitor) *ClientOptions {
 	c.PoolMonitor = m
 	return c
 }
 
-// SetMonitor specifies a command monitor used to see commands for a client.
+// SetMonitor specifies a CommandMonitor to receive command events. See the event.CommandMonitor documentation for more
+// information about the structure of the monitor and events that can be received.
 func (c *ClientOptions) SetMonitor(m *event.CommandMonitor) *ClientOptions {
 	c.Monitor = m
 	return c
 }
 
-// SetReadConcern specifies the read concern.
+// SetReadConcern specifies the read concern to use to configure the Client. A read concern level can also be set
```

Comment:
You start using the phrase "to configure the Client" starting here.  It seems redundant given that these are client options.  Re-phrase here and below.  Perhaps just:

   SetReadConcern specifies the read concern to use for read operations

Reply:
done.